accepted/wasmtime-compile-time-builtins.md

```diff
@@ -292,9 +292,14 @@ and 64-bit variant of their builtins. The `native` mnemonic hints that the
 memory operation is operating on the native memory address space, not a
 particular Wasm memory, and uses native endianness.
 
-The final intrinsic, `resource.address`, gives the address of the host data in
-the resource table for a particular resource handle. It will raise a trap on
-invalid resources and out-of-bounds resource table accesses.
+The final intrinsic, `store.data_address`, gives a `*mut T` pointer containing
+the address of the `T` host data in the `wasmtime::Store<T>`. As long as
+embedders ensure that their `T` type is `#[repr(C)]`, then their compile-time
```

Comment:
you also have to watch out for primitive type alignment changing offsets, e.g. in `#[repr(C)] struct S(u8, f64)` the `f64` field is at offset `4` on `i686-unknown-linux-gnu` but at offset `8` on `x86_64-unknown-linux-gnu`

Reply:
That's true and I expect we will need to add some wording to our documentation for this feature around what exactly the safety conditions will need to be, such that the compile-time builtins can portably access the host data. My plan right now is to document these things in detail as I prototype and dog-food the feature.